[CI] MixedClusterClientYamlTestSuiteIT {p0=cluster.component_template/10_basic/Basic CRUD} failing in release tests

This test is newly added and has been consistently failing in our [release tests build](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/).

```
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=cluster.component_template/10_basic/Basic CRUD} FAILED
    java.lang.AssertionError: Failure at [cluster.component_template/10_basic:39]: expected [404] status code but api [cluster.get_component_template] returned [400 Bad Request] [{"error":"no handler found for uri [/_component_template/test?error_trace=true] and method [GET]"}]
    Expected: <404>
         but: was <400>
        at __randomizedtesting.SeedInfo.seed([E7C3409F53BCFC4D:6F977F45FD4091B5]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:382)
```

https://gradle-enterprise.elastic.co/s/bldz2t6m5pzfm/tests/xrntlbe6hxtlm-3b6npyoqalafk

@dakrone  since you recently added this test do you mind taking a look?

--- a/build.gradle
+++ b/build.gradle
@@ -547,3 +547,14 @@ allprojects {
     }
   }
 }
+
+// TODO: remove this once 7.7 is released and the 7.x branch is 7.8
+subprojects {
+  pluginManager.withPlugin('elasticsearch.testclusters') {
+    testClusters.all {
+      if (org.elasticsearch.gradle.info.BuildParams.isSnapshotBuild() == false) {
+        systemProperty 'es.itv2_feature_flag_registered', 'true'
+      }
+    }
+  }
+}

--- a/distribution/archives/integ-test-zip/build.gradle
+++ b/distribution/archives/integ-test-zip/build.gradle
@@ -1,4 +1,3 @@
-import org.elasticsearch.gradle.info.BuildParams
 /*
  * Licensed to Elasticsearch under one or more contributor
  * license agreements. See the NOTICE file distributed with
@@ -31,11 +30,5 @@ integTest.runner {
       "${-> testClusters.integTest.singleNode().getServerLog()}"
   } else {
     systemProperty 'tests.logfile', '--external--'
-  }
-}
-
-testClusters.integTest {
-  if (BuildParams.isSnapshotBuild() == false) {
-    systemProperty 'es.itv2_feature_flag_registered', 'true'
   }
 }

--- a/qa/smoke-test-multinode/build.gradle
+++ b/qa/smoke-test-multinode/build.gradle
@@ -17,8 +17,6 @@
  * under the License.
  */
 
-import org.elasticsearch.gradle.info.BuildParams
-
 apply plugin: 'elasticsearch.testclusters'
 apply plugin: 'elasticsearch.standalone-rest-test'
 apply plugin: 'elasticsearch.rest-test'
@@ -46,11 +44,5 @@ integTest.runner {
       'cat.templates/10_basic/Sort templates',
       'cat.templates/10_basic/Multiple template',
     ].join(',')
-  }
-}
-
-testClusters.integTest {
-  if (BuildParams.isSnapshotBuild() == false) {
-    systemProperty 'es.itv2_feature_flag_registered', 'true'
   }
 }

--- a/server/src/main/java/org/elasticsearch/action/ActionModule.java
+++ b/server/src/main/java/org/elasticsearch/action/ActionModule.java
@@ -378,9 +378,6 @@ public class ActionModule extends AbstractModule {
 
     static {
         final String property = System.getProperty("es.itv2_feature_flag_registered");
-        if (Build.CURRENT.isSnapshot() && property != null) {
-            throw new IllegalArgumentException("es.itv2_feature_flag_registered is only supported in non-snapshot builds");
-        }
         if (Build.CURRENT.isSnapshot() || "true".equals(property)) {
             ITV2_FEATURE_FLAG_REGISTERED = true;
         } else if ("false".equals(property) || property == null) {

--- a/x-pack/qa/core-rest-tests-with-security/build.gradle
+++ b/x-pack/qa/core-rest-tests-with-security/build.gradle
@@ -1,5 +1,3 @@
-import org.elasticsearch.gradle.info.BuildParams
-
 apply plugin: 'elasticsearch.testclusters'
 apply plugin: 'elasticsearch.standalone-rest-test'
 apply plugin: 'elasticsearch.rest-test'
@@ -38,10 +36,4 @@ testClusters.integTest {
 
   user username: System.getProperty('tests.rest.cluster.username', 'test_user'),
     password: System.getProperty('tests.rest.cluster.password', 'x-pack-test-password')
-}
-
-testClusters.integTest {
-  if (BuildParams.isSnapshotBuild() == false) {
-    systemProperty 'es.itv2_feature_flag_registered', 'true'
-  }
 }

